tokio-trace-fmt/src/lib.rs

```diff
@@ -24,12 +26,32 @@ mod span;
 pub use filter::Filter;
 pub use span::Context;
 
+/// A type that can format a tracing `Event` for a `fmt::Write`.
+///
+/// `Formatter` is primarily used in the context of [`FmtSubscribe`]. Each time an event is
+/// dispatched to [`FmtSubscribe`], the subscriber forwards it to its associated `Formatter` to
+/// emit a log message.
+///
+/// This trait is already implemented for function pointers with the same signature as `format`.
+pub trait Formatter<N> {
```

Comment:
WDYT of calling this `FmtEvent` or `FormatEvent`, rather than `Formatter`?

If we want to introduce log lines on span entry/creation later, I think it would be good to distinguish a trait for formatting `Event`s from a trait for formatting `Span`s. Also, I'd rather not conflict with `std::fmt::Formatter`. 

Finally, this is less important, but most of the other trait names in this library are verbs rather than nouns...